wasi/2023/WASI-09-21.md

```diff
@@ -30,3 +30,180 @@ The meeting will be on a zoom.us video conference.
+
+**Till**: Most of the stuff specified in the WHATWG spec, I implemented at Fastly’s Compute@Edge a couple of years ago. One of the goals of the wasi-http spec is to be compatible with this. Over the last few days, I have started working on this within JavaScript side and have not seen any major users. I expect this to pass the WPT platform web test suite. I don’t see any issues there and feel very optimistic about it.
+
+**Petr Penzin**: Nice, thank you.
+
+**SBC**: A lot of the other WASI API’s do not try to mirror web-y things. Do you think this is going to be a requirement to mirror the web everywhere? Why bother to mention browsers?
+
+**Luke Wagner**: People are already polyfilling components into browsers. The goal is to be able to run components in lots of places and Service Workers.
+
+**SBC**: Will all WASI API’s mirror web API’s.
+
+**Luke Wagner**: Not in a blank check sense. OPFS, it would be cool to implement wasi-fs on OPFS via polyfill.
+
+**Till**: more accurate to say fetch spec is implemented in terms of the http spec as is the wasi-http spec. The fetch implementation I just mentioned is non-trivial. It has a lot of code in it to implement the fetch spec in terms of another API. It is not an identical API but serves roughly the same function.
+
+**Petr Penzin**: OPFS, when not in the browser but running in JS engines, then you have similar API’s but not exactly the same. The goal of compiling C++, Rust, what have you, then these API’s will not exactly match the JS APIs. It makes sense that there is a balance here.
+
+**Piotr Sikora**: Will this be phase criteria for all WASI proposals or specific to HTTP? 
+
+**Pat Hickey:** No, specific to wasi-http. We think this is appropriate for wasi-http since fetch API exists and so it makes it desirable. Sockets for instance, there will never be a requirement for it to work on the web, there will never be a way to make it work on the web. 
+
+**Piotr Sikora**: What about WebTransport and/or WebSockets?
+
+**Pat Hickey:** Exposing web transport would be very interesting.
+
+**Piotr Sikora**: It could also lead to partitioning of the WASI worlds. We want to create the HTTP proxy world and that requirement may conflict with the needs of browsers. 
+
+**Pat Hickey**:  We expect that the http-proxy world is the thing that will work in browsers. Sockets is very intentionally not part of this world. One world we’ve tossed around conversationally is a caching http proxy store. When in a serverless environment, you could have more sharing of that cache.
+
+**Till**: @Piotr Sikora part of the stage 2 entrance criteria is defining stage 4 criteria, so it's not purely up to the proposal champions. Instead, the WASI SG has to assess the validity of those criteria in deciding whether the proposal should advance to stage 2
+
+**Petr Penzin**: Easier to see which parts of the web are mappable and which are not. Since WASI does not make this a hard requirement. 
```

Comment:
```suggestion
**Petr Penzin**: If there was a requirement to consider Web mapping, it would be easier to see which parts of the web are mappable and which are not. Though that might not make sense for some proposals. 
```